src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -95,7 +95,7 @@ public override void Initialize(AnalysisContext analysisContext)
                             foreach (INamedTypeSymbol symbol in externallyVisibleNamedTypes)
                             {
                                 string symbolName = symbol.Name;
-                                if (s_wellKnownSystemNamespaceTable.ContainsKey(symbolName))
+                                if (s_wellKnownSystemNamespaceTable!.ContainsKey(symbolName))
```

Comment:
💡 This suppression can be replaced with a wrapper property that initializes the field on first access.